tests/api_tests/comments/test_views.py

```diff
@@ -0,0 +1,889 @@
+from nose.tools import *  # flake8: noqa
+
+from api.base.settings.defaults import API_BASE
+
+from tests.base import ApiTestCase
+from tests.factories import ProjectFactory, AuthUserFactory, CommentFactory, RegistrationFactory
+
+
+class TestCommentDetailView(ApiTestCase):
+    def setUp(self):
+        super(TestCommentDetailView, self).setUp()
+        self.user = AuthUserFactory()
+        self.contributor = AuthUserFactory()
+        self.non_contributor = AuthUserFactory()
+
+        self.private_project = ProjectFactory(is_public=False, creator=self.user)
+        self.private_project.add_contributor(self.contributor, save=True)
+        self.public_project = ProjectFactory(is_public=True, creator=self.user)
+        self.public_project.add_contributor(self.contributor, save=True)
+        self.registration = RegistrationFactory(creator=self.user)
+
+        self.comment = CommentFactory(node=self.private_project, target=self.private_project, user=self.user)
+        self.public_comment = CommentFactory(node=self.public_project, target=self.public_project, user=self.user)
+        self.registration_comment = CommentFactory(node=self.registration, user=self.user)
```

Comment:
Ah, correction: the page was borked. The functionality and UI to comment on registrations does exist, so yay tests!